src/Diagnostics.ModelsAndUtils/Models/ResponseExtensions/Card.cs

```diff
@@ -0,0 +1,138 @@
+            Descriptions = descriptions;
+            Icon = icon;
+            ActionType = actionType;
+            ActionValue = actionValue;
+        }
+    }
+
+    public static class ResponseCardExtension
+    {
+        /// <summary>
+        /// Adds a list of Cards to Response
+        /// </summary>
+        /// <param name="response">Response</param>
+        /// <param name="cards">List<![CDATA[<Card>]]></param>
+        /// <returns></returns>
+        /// <example> 
+        /// This sample shows how to use <see cref="AddCards"/> method.
+        /// <code>
+        /// public async static Task<![CDATA[<Response>]]> Run(DataProviders dp, OperationContext cxt, Response res)
+        /// {
+        ///     var cards = new List<![CDATA[<Card>]]>();
+        ///     cards.Add(new Card(
+        ///         {
+        ///             Title = "HTTP 4XX Errors",
+        ///             ShortDescription = "Select this category for following issue types",
+        ///             Descriptions = new List<![CDATA[<string>]]> { "Getting 401, 405, 403 error","The web app is stopped","You are not authorized to view this page" },
+        ///             Icon = "fa-circle",
+        ///             ActionsType =  CardActionType.Detector,
+        ///             ActionValue = "http4xx"
+        ///         }
+        ///     ));
+        ///
+        ///     cards.Add(new Card(
+        ///         {
+        ///             Title = "HTTP 5XX Errors",
+        ///             ShortDescription = "Chosse these if app is failing with these errors",
+        ///             Descriptions = new List<![CDATA[<string>]]> { "App failing with 502, 503, 500 error","The Page cannot be displayed","App failing with a server error" },
+        ///             Icon = "fa-exclamation",
+        ///             ActionsType =  CardActionType.Detector,
+        ///             ActionValue = "httpservererrors"
+        ///         }
+        ///     ));
+        ///
+        ///     res.AddCards(cards);
+        /// }
+        /// </code>
+        /// </example>
+        public static DiagnosticData AddCards(this Response response, List<Card> cards)
+        {
+            var table = new DataTable();
+            table.Columns.Add(new DataColumn("Title", typeof(string)));
+            table.Columns.Add(new DataColumn("Icon", typeof(string)));
+            table.Columns.Add(new DataColumn("ShortDescription", typeof(string)));
+            table.Columns.Add(new DataColumn("Descriptions", typeof(string)));
+            table.Columns.Add(new DataColumn("ActionType", typeof(string)));
+            table.Columns.Add(new DataColumn("ActionValue", typeof(string)));
+
+            cards.ForEach(card =>
+            {
+                table.Rows.Add(new object[] {
+                    card.Title,
+                    card.Icon,
+                    card.ShortDescription,
+                    JsonConvert.SerializeObject(card.Descriptions),
```

Comment:
You may not have to serialize these values, because the rows is dynamic[,] now which means they would be allowed. When you do a makeset in kusto for example, it returns an array, not a string serialized array. Can you try and see what the response looks like without serializing them?

Reply:
Ignore this comment, I was wrong, it did not serialize correctly on its own.

Reply:
Yea that's what I thought that serialization is needed ...